### DISCUSSION TITLE
Is WinMerge affected by CVE-2021-4428 (log4j)?

### DISCUSSION THREAD

**Discussion by horsetalk** [WinMerge/winmerge#1083] [Q&A]
Can you confirm that WinMerge is not impacted by CVE-2021-4428 (Log4j vulnerability)?

**Comment by sdottaka** [comment]
WinMerge does not use log4j, which is a Java library, because it is basically written in C++.

The exception is the Apache tika plugin, which uses log4j internally.

The version of log4j used is 1.x, not the vulnerable version 2. Also, it does not appear to output any logs under normal use, so we do not see a problem.

https://issues.apache.org/jira/browse/TIKA-3616?jql=project%20%3D%20TIKA%20AND%20text%20~%20%22log4j%22
